src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -584,15 +584,15 @@ export class ObjectExplorerService implements IObjectExplorerService {
 		return this.expandOrRefreshTreeNode(session, parentTree, needsRefresh);
 	}
 
-	public refreshTreeNode(session: azdata.ObjectExplorerSession, parentTree: TreeNode): Promise<TreeNode[]> {
-		return this.expandOrRefreshTreeNode(session, parentTree, true);
+	public async refreshTreeNode(session: azdata.ObjectExplorerSession, parentTree: TreeNode): Promise<TreeNode[]> {
+		return await this.expandOrRefreshTreeNode(session, parentTree, true);
```

Comment:
I don't get how this is changing anything - returning a promise is essentially the same as awaiting the promise in an async function and then returning the value (which is then wrapped in a Promise itself).

Let me take a look at how this code is used...unfortunately I'm still not able to repro this myself.

Reply:
To reproduce:
Make 4-5 diff connections, reload window/restart ADS and open all 5 OE connections at the same time.